plugins/maintenance/src/main/java/org/apache/cloudstack/maintenance/ManagementServerMaintenanceManagerImpl.java

```diff
@@ -294,19 +343,21 @@ public ManagementServerMaintenanceResponse prepareForShutdown(PrepareForShutdown
             throw new CloudRuntimeException("Management server is not in the right state to prepare for shutdown");
         }
 
+        final List<ManagementServerHostVO> preparingForMaintenanceOrShutDownMsList = msHostDao.listBy(State.PreparingForMaintenance, State.PreparingForShutDown);
+        if (CollectionUtils.isNotEmpty(preparingForMaintenanceOrShutDownMsList)) {
+            throw new CloudRuntimeException("Cannot prepare for shutdown, there are other management servers preparing for maintenance/shutdown");
+        }
```

Comment:
new method

Reply:
no changes, validation should be ok in the method

Reply:
done